pkg/search/result.go

```diff
@@ -220,6 +225,13 @@ func (issue Issue) ExportData(fields []string) map[string]interface{} {
 				"name":          name,
 				"nameWithOwner": nameWithOwner,
 			}
+		case "state":
+			if issue.PullRequest.Merged() {
+				data[f] = "merged"
+			} else {
+				// either an issue or a pull request that is not merged
+				data[f] = issue.State
+			}
```

Comment:
The state property of a pull request is shown in multiple places, including JSON export. To make sure everyone uses the same unified logic, could this be extracted into a method `State() string`?

The old struct field `` State string `json:"state"` `` could be renamed into something that discourages its direct access in favor of the new `State()` method.

Reply:
@mislav done in https://github.com/cli/cli/pull/6411/commits/06bb157858322c9c37de4cf5e3f5c16fc38c57a2

I hope the code comment I added to the function is clear and expresses the intent correctly 🤞